packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts

```diff
@@ -339,7 +363,7 @@ export class RedirectsMiddleware extends MiddlewareBase {
    * @param {string} [params.locale] - The locale prefix to include in the URL if present.
    * @returns {string | undefined} - return query string if any of the query permutations match the provided pattern, undefined otherwise.
    */
-  private isPermutedQueryMatch({
+  private getPermutedQueryMatch({
```

Comment:
This method, which generates all permutations of query parameters to test against regex patterns, can lead to performance degradation due to its factorial time complexity (O(n!)). Can this method/approach be refactored to avoid the performance degradation that leads to timeouts in all hosting envs (including netlify and vercel). This would address #2004

Reply:
This method is removed and there added another logic